### PR TITLE
Feature module: aws-iam-policy

### PR DESCRIPTION
**Pull request by nsarupr**
# Description
New module: aws-iam-policy

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

Example on how to use the IAM policy module:
configuration.yaml
```yaml
name: <env_name>
org_name: <org_name>
providers:
  aws:
    region: "us-east-1"
    account_id: 000000000000
modules:
  - type: base
  - type: k8s-cluster
  - type: k8s-base
  - type: aws-iam-policy
    file: policy.json
```
policy.json
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "ec2:Describe*"
      ],
      "Effect": "Allow",
      "Resource": "*"
    }
  ]
}
```
## How has this change been tested, beside unit tests?
Tested Manually.

### CODE REVIEW

File: modules/aws_iam_policy/aws-iam-policy.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+description: Creates an IAM policy
+---
+
+This module can be used to create and manage an AWS IAM policy via opta
+
+### Example
+
+```
+  - name: deployer
+    type: aws-iam-role
```
Comment (juandiegopalomino) [inline modules/aws_iam_policy/aws-iam-policy.md:16]: ?
this is not an example of aws iam policy

File: modules/aws_iam_policy/tf_module/main.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+resource "random_string" "policy_suffix" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+resource "aws_iam_policy" "policy" {
+  name   = "${var.env_name}-${var.layer_name}-${var.module_name}"
+  policy = file(var.file)
```
Comment (juandiegopalomino) [inline modules/aws_iam_policy/tf_module/main.tf:9]: we didn't need to do anything fancy for the relative file path, right? (check other examples)

File: modules/aws_iam_policy/aws-iam-policy.md (hunk 1 above)
Comment (nsarupr) [reply modules/aws_iam_policy/aws-iam-policy.md:16]: My bad, let me add a proper example for the same.

File: modules/aws_iam_policy/aws-iam-policy.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+This module can be used to create and manage an AWS IAM policy via opta
+
+### Example
+
+```
+  - name: deployer
+    type: aws-iam-policy
+    file: valid_policy.json
```
[the remark this answers is not in the document]
Comment (nsarupr) [reply modules/aws_iam_policy/aws-iam-policy.md:16]: Added an example in the PR as well as in the readme.

**Review by juandiegopalomino — APPROVED** [review]

**Pull request merged**